### ISSUE TITLE
Incorrect dependency scope for plume-util in Checker dataflow POM file

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#4612]
I was just working on trying to update NullAway to use version 3.12.0 of the dataflow artifact.  The POM for dataflow indicates its dependency on plume-util is in the runtime scope:

https://search.maven.org/artifact/org.checkerframework/dataflow/3.12.0/jar (see the XML)

Using the runtime scope will not add plume-util to the compile classpath for a library dependent on dataflow.  But, the `org.checkerframework.dataflow.cfg.node.Node` class implements the `org.plumelib.util.UniqueId` interface from plume-util, so clients of the dataflow library may need to also have plume-util in their compile classpath (I am getting a compile error otherwise).  I can work around this by manually adding the plume-util dependence to NullAway, but probably the generated POM file should be fixed.

**Comment by mernst** [comment]
Another possibility would be to rewrite the dataflow framework to eliminate the dependency on plume-util.  In addition to `UniqueId`, there are also uses of `StringsPlume` and `CollectionsPlume`.

**Comment by msridhar** [comment]
> Another possibility would be to rewrite the dataflow framework to eliminate the dependency on plume-util. In addition to `UniqueId`, there are also uses of `StringsPlume` and `CollectionsPlume`.

I don't think that is necessary at least for this issue.  The fix may be as simple as changing this line to be a `compile` dependency rather than `implementation`:

https://github.com/typetools/checker-framework/blob/79b4b4f96bae8b3c501e6e7a7b6bf99cc9eb8438/dataflow/build.gradle#L4

I checked and with that change the dependency uses the `compile` scope in the generated POM, as desired.  However, the change should be tested for interactions with the shadow plugin.  Also, support for the `compile` dependency type is deprecated and has been removed in Gradle 7.  It may be good to switch to the `java-library` plugin and then use the more modern `api` dependency type.
